lib/services/profiles-api-service.js

```diff
@@ -102,41 +102,80 @@ function profileApiServiceFactory(
 
                 });
             } else {
-                return self.createNodeAndRunDiscovery(macAddresses);
+                return self.createNodeAndRunDiscovery(macAddresses, options);
             }
         });
     };
 
-    ProfileApiService.prototype.runDiscovery = function(node) {
+    ProfileApiService.prototype.runDiscovery = function(node, options) {
         var self = this;
+        var configuration;
 
+        if (node.type === 'switch') {
+            configuration = self.getSwitchDiscoveryConfiguration(node, options.switchVendor);
+        } else {
+            configuration = {
+                name: 'Graph.SKU.Discovery',
+                options: {
+                    defaults: {
+                        graphOptions: {
+                            target: node.id
+                        },
+                        nodeId: node.id
+                    }
+                }
+            };
+        }
+
+        // The nested workflow holds the lock against the nodeId in this case,
+        // so don't add it as a target to the outer workflow context
+        return workflowApiService.createAndRunGraph(configuration, null)
+        .then(function() {
+            return self.waitForDiscoveryStart(node.id);
+        })
+        .then(function() {
+            return node;
+        });
+    };
+
+    ProfileApiService.prototype.getSwitchDiscoveryConfiguration = function(node, vendor) {
         var configuration = {
-            name: 'Graph.SKU.Discovery',
+            name: 'Graph.SKU.Switch.Discovery.Active',
             options: {
                 defaults: {
                     graphOptions: {
                         target: node.id
                     },
                     nodeId: node.id
+                },
+                'vendor-discovery-graph': {
+                    graphName: null
                 }
             }
         };
 
-        return workflowApiService.createAndRunGraph(configuration)
-        .then(function() {
-            return self.waitForDiscoveryStart(node.id);
-        })
-        .then(function() {
-            return node;
-        });
+        vendor = vendor.toLowerCase();
+
+        if (vendor === 'cisco') {
```

Comment:
@jlongever If I understand you correctly, passing the filename in directly is what my current proposal is doing (cisco-specific cataloging graph: https://github.com/benbp/on-taskgraph/blob/cisco-poap/lib/graphs/cisco-switch-discovery-graph.js#L18). Do you mean keeping a mapping somewhere of vendor type to a collection of script names?

@pscharla yeah, that's what I was talking about. With the deploy-cisco graph though, I put that in as an example, but I imagine we'll probably want to do less standardization on the dynamic SKU graphs, since some users may want to do different things (only deploy startup config, also deploy boot images, etc.).